update.sh

```diff
@@ -1,10 +1,23 @@
 #!/bin/bash
+
 INSTALL_DIR="/opt/zpui"
 SUDO=''
+
+cd "$(dirname "$0")"  # Make sure we are in the script's directory when running
+set -e  # Strict mode : script stops if any command fails
+
+
 if (( $EUID != 0 )); then
     SUDO='sudo'
 fi
-git pull
-$SUDO rsync -av --delete ./  $INSTALL_DIR
-$SUDO systemctl restart zpui.service
 
+git pull origin master  # Always tell the branch we pull from
+
+${SUDO} pip install -r requirement.txt  # Make sure we have the latest dependencies installed
+
+# Run tests
+pytest2 --doctest-modules -v --doctest-ignore-import-errors --ignore=apps/example_apps/fire_detector/ --ignore=ui/tests/test_checkbox.py  #  todo : fixes checkbox testing not working at the moment
```

Comment:
I suggest adding a "-f" command-line switch overriding the test run, it would be helpful to users that want to upgrade "no matter what" =)

Reply:
done